<launch>    
    <arg name="manager_name" default="nodelet_manager" />
    <arg name="manager_threads" default="4" />


    <arg name="should_output_marker"        default="true"/>
    <arg name="tracking_method"             default="1"/>
    <!-- Particle Filter base : 1, Particle Filter + Support Vector Data Description : 2 -->

    <arg name="use_omni"                    default="false"/>

    <arg name="use_rotate"                  default="true"/>

    <!-- rviz -->
    <arg name="rviz"                    default="true"/>
    <arg name="rviz_cfg"                default="$(find person_follower4)/config/rviz/person_follower4.rviz"/>
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_cfg)"/>
    
    <group ns = "person_follower">
        <node pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen">
            <param name="num_worker_threads" value="$(arg manager_threads)" />
        </node>

        <!-- ssd_node -->
        <include file="$(find person_follower4)/launch/include/object_detect.launch.xml">
        </include>

        <node pkg="nodelet" type="nodelet" name="particle_person_tracker" args="load person_follower4/PersonTracker $(arg manager_name)" >
            <rosparam file="$(find person_follower4)/param/topic_params.yaml" command="load" />
            <param name="tracking_method"       type="int"      value="$(arg tracking_method)"/>
        </node>

        <node pkg="nodelet" type="nodelet" name="dwa_path_generator" args="load person_follower4/PathGenerator $(arg manager_name)" >
            <rosparam file="$(find person_follower4)/param/dwa_params.yaml" command="load" />
            <rosparam file="$(find person_follower4)/param/path_generator_params.yaml" command="load" />
            <param name="use_omni"              type="bool"         value="$(arg use_omni)"/>
            <param name="is_display_path"       type="bool"         value="$(arg should_output_marker)"/>
            <param name="is_display_all_path"   type="bool"         value="$(arg should_output_marker)"/>
            <!-- <remap from="/person_follower/follow_position" to="/follow_position" /> -->
        </node>
        <remap from="/person_follower/velocity" to="/mobile_base/commands/velocity"/>

        <!-- velocity_smoother -->
        <!-- <include file="$(find person_follower4)/launch/include/velocity_smoother.launch.xml"> -->
        <!-- </include> -->

        <node pkg="nodelet" type="nodelet" name="sensor_rotator" args="load person_follower4/PersonAimSensorRotator $(arg manager_name)" >
            <rosparam file="$(find person_follower4)/param/sensor_rotator_params.yaml" command="load" />
            <param name="use_rotate"            type="bool"     value="$(arg use_rotate)"/>
            <!-- <remap from="/person_follower/rotate_position" to="/rotate_position" /> -->
        </node>

        <group ns = "$(arg manager_name)">
            <rosparam file="$(find person_follower4)/param/particle_person_tracker_params.yaml" command="load" />
            <param name="should_output_marker" type="bool"      value="$(arg should_output_marker)"/>
        </group>

    </group>

</launch>